packages/api-client-core/src/GadgetConnection.ts

```diff
@@ -100,7 +99,7 @@ export class GadgetConnection {
     } else {
       this._fetchImplementation = fetchPolyfill;
     }
-    this.websocketImplementation = options.websocketImplementation ?? WebSocket;
+    this.websocketImplementation = options.websocketImplementation ?? globalThis?.WebSocket ?? WebSocket;
```

Comment:
I also upgraded the isomorphic-ws library, which stopped monkeypatching the global. This makes our tests keep working.